musl doesn't GLOB_TILDE, so simply don't use it here.

diff --git a/lobster/src/platform.cpp b/lobster/src/platform.cpp
index ceaec49..68e2443 100644
--- lobster/src/platform.cpp
+++ lobster/src/platform.cpp
@@ -46,6 +46,10 @@
 #include "sdlincludes.h"  // FIXME
 #endif
 
+#ifndef GLOB_TILDE
+#define GLOB_TILDE 0
+#endif
+
 // Main dir to load files relative to, on windows this is where lobster.exe resides, on apple
 // platforms it's the Resource folder in the bundle.
 string datadir;

